openpgp/src/main/java/com/yubico/yubikit/openpgp/OpenPgpSession.java

```diff
@@ -0,0 +1,1283 @@
+
+            }
+            throw e;
+        }
+        Logger.info(logger, "New {} PIN set", pw);
+    }
+
+    /**
+     * Change the User PIN.
+     *
+     * @param pin    the current User PIN
+     * @param newPin the new User PIN to set
+     * @throws ApduException       in case of an error response from the YubiKey
+     * @throws IOException         in case of connection error
+     * @throws InvalidPinException in case of the wrong PIN in case of the wrong PIN
+     */
+    public void changeUserPin(String pin, String newPin) throws ApduException, IOException, InvalidPinException {
+        changePw(Pw.USER, pin, newPin);
+    }
+
+    /**
+     * Change the Admin PIN.
+     *
+     * @param pin    the current Admin PIN
+     * @param newPin the new Admin PIN to set
+     * @throws ApduException       in case of an error response from the YubiKey
+     * @throws IOException         in case of connection error
+     * @throws InvalidPinException in case of the wrong PIN
+     */
+    public void changeAdminPin(String pin, String newPin) throws ApduException, IOException, InvalidPinException {
+        changePw(Pw.ADMIN, pin, newPin);
+    }
+
+    /**
+     * Set the Reset Code for User PIN.
+     * <p>
+     * The Reset Code can be used to set a new User PIN if it is lost or becomes
+     * blocked, using the reset_pin method.
+     * <p>
+     * This command requires Admin PIN verification.
+     *
+     * @param resetCode the Reset Code to set
+     * @throws ApduException in case of an error response from the YubiKey
+     * @throws IOException   in case of connection error
+     */
+    public void setResetCode(String resetCode) throws ApduException, IOException {
+        Logger.debug(logger, "Setting a new PIN Reset Code");
+        byte[] data = getKdf().process(Pw.RESET, resetCode);
+        putData(Do.RESETTING_CODE, data);
+        Logger.info(logger, "New Reset Code has been set");
+    }
+
+    /**
+     * Resets the User PIN in case it is lost or blocked.
+     * <p>
+     * This can be done either after performing Admin PIN verification, or by providing the Reset
+     * Code.
+     * <p>
+     * This command requires Admin PIN verification, or the Reset Code.
+     *
+     * @param newPin    the new User PIN to set
+     * @param resetCode the Reset Code, which is needed if the Admin pin has not been verified
+     * @throws ApduException       in case of an error response from the YubiKey
+     * @throws IOException         in case of connection error
+     * @throws InvalidPinException in case of the wrong PIN
+     */
+    public void resetPin(String newPin, @Nullable String resetCode) throws
+            ApduException, IOException, InvalidPinException {
+        Logger.debug(logger, "Resetting User PIN");
+        byte p1 = 2;
+        Kdf kdf = getKdf();
+        byte[] data = kdf.process(Pw.USER, newPin);
+        if (resetCode != null) {
+            logger.debug("Using Reset Code");
```

Comment:
There are a few instances or directly using `logger` instead of wrapping it in `Logger`.